libs/k8s-client/src/KubernetesApiClientFacade.php

```diff
@@ -237,7 +237,7 @@ public function deleteAllMatching(?DeleteOptions $deleteOptions = null, array $q
      *         never))))))
      */
     // phpcs:ignore Generic.Files.LineLength.MaxExceeded
-    private function getApiForResource(string $resourceType): ConfigMapsApiClient|EventsApiClient|PersistentVolumeClaimApiClient|PodsApiClient|SecretsApiClient
+    public function getApiForResource(string $resourceType): ConfigMapsApiClient|EventsApiClient|PersistentVolumeClaimApiClient|PodsApiClient|SecretsApiClient
```

Comment:
bude se mi hodit v sandboxech

Reply:
jakym zpusobem? nestaci ti stavajici univerzalni metody (createModels()/deleteModels()/...) nebo accessory na konkretni APIcka (pods()/secrets()/...).

Reply:
Chtěl jsem dělat co nejméně zásahů do:

checkResourceExists + deleteResourcesByLabels

https://github.com/keboola/sandboxes/blob/b32ffd3b1b5c962d7475ae97fb980110b3c2f6ae/src/Service/Kubernetes/Handler.php#L94-L116

Pokud tahle metoda nebude public, tak  budu muset její vnitřní logiku z části ukradnout a zduplikovat v těch sandboxech

Reply:
Jo uz to vidim, tam se optionally vynechava `pvc`. Porad se mi to moc nelibi :slightly_smiling_face: ale fakt je, ze me ted nenapada, jak to jinak snadni vyresit.

Premyslel jsem nad pridanim nejakyho `$onlyResrouceTypes` optional argumentu `deleteAllMatching`, kde by si vyjmenoval, jaky typy chec mazat (vice/mene tak, jak se to ted vlastne delat v sandboxes). K ty metode by to celkem pekne sedelo (protoze jinak ji vlastne bdue reimplementovat presne s tou moznosti konfigurace :slightly_smiling_face:). Jen si nejsem jistej, jak to ublizi citelnosti, kdyz bude mit treti argument takle.

Jinak proc se tomu otevreni branim - otvira to novou cestu jak delat neco, co uz ta facade umoznuje. Cimz se stane to, ze nekde v kodu budes mit `$facade->pods()->...`, nekde `$facade->getApiForResource(Pod::class)->...` a budes resit, proc a jaky je v tom rozdil, nez dojdes k zaveru, ze tam vlastne zadny neni a je to jen stejne volani toho sameho.

Reply:
Asi bych `deleteAllMatching` predelal, aby mela stejny api jako listMatching. To ze to uvnitr ma ted nejaky omezeni mi prijde divny a nemelo by byt soucasti toho klienta.